icon/iiss/calculator/prep.go

```diff
@@ -366,27 +352,26 @@ func (p *PRepInfo) InitAccumulated() {
 func (p *PRepInfo) ApplyVote(vType VoteType, votes icstage.VoteList, offset int) {
 	for _, vote := range votes {
 		key := icutils.ToKey(vote.To())
-		if prep, ok := p.preps[key]; !ok {
-			continue
-		} else {
-			prep.ApplyVote(vType, vote.Amount(), p.offsetLimit-offset)
-			p.log.Debugf("ApplyVote %+v: by %d, %d %+v, %d * %d",
-				prep, vType, offset, vote, vote.Amount(), p.offsetLimit-offset)
-			p.preps[key] = prep
+		prep, ok := p.preps[key]
+		if !ok {
+			prep = p.Add(vote.To(), icmodule.ESDisablePermanent, new(big.Int), new(big.Int), 0, false)
```

Comment:
Logic is changed to add PRep structure if it doesn't exist

Reply:
Previous code had a bug.

This code works when ICONist delegate to not registered P-Rep.

If ICONist delegate to not registered P-Rep, they can't get rewards. After that P-Rep register, they must get rewards with past delegation. So we must store information of that P-Rep.